src/utils/segments.ts

```diff
@@ -32,3 +44,61 @@ export function getSegments(encoder: Tiktoken, inputText: string) {
 
   return segments;
 }
+
+export function getHuggingfaceSegments(
+  tokenizer: PreTrainedTokenizer,
+  inputText: string
+): Segment[] {
+  const tokens = tokenizer.encode(inputText);
+  // remove first token, which is always <s>?
+  tokens.shift();
+  console.log("HF tokens", tokens);
+  const te = new TextEncoder();
+  const segments: { text: string; tokens: { id: number; idx: number }[] }[] =
+    [];
+
+  let byteAcc: number[] = [];
+  let tokenAcc: { id: number; idx: number }[] = [];
+  let inputGraphemes = graphemer.splitGraphemes(inputText);
+
+  const tokenStrs = tokenizer.model.convert_ids_to_tokens(tokens);
+  const dec = tokenizer.decoder ?? ((x: string) => x);
+  console.log("decoder", dec, dec.config);
+  return tokenStrs.map((token, idx) => ({
+    // This is extremely dumb and I hate it. Oh, sentencepiece!
+    text: idx == 0 ? dec([token]) : dec([" ", token]),
```

Comment:
Will this work if meta symbol is not before the token?

Reply:
This is what decoder looks like for llama-7b:
<img width="501" alt="Screenshot 2024-03-15 at 10 05 46 AM" src="https://github.com/dqbd/tiktokenizer/assets/355540/3c9cf40b-58e8-435e-8aa3-6e3034147207">

phi-2
<img width="775" alt="Screenshot 2024-03-15 at 10 13 59 AM" src="https://github.com/dqbd/tiktokenizer/assets/355540/d4b16c30-cd10-4cee-ae65-894a2037359a">